src/frontend/src/pages/AdminToolsPage/BOMConfig/ManufacturerTable.tsx

```diff
@@ -1,35 +1,102 @@
-import { TableRow, TableCell, Box } from '@mui/material';
+import DeleteIcon from '@mui/icons-material/Delete';
+import { Box, Grid, IconButton, TableCell, TableRow } from '@mui/material';
 import LoadingIndicator from '../../../components/LoadingIndicator';
+import { NERButton } from '../../../components/NERButton';
+import { useDeleteManufacturer, useGetAllManufacturers } from '../../../hooks/bom.hooks';
+import { useToast } from '../../../hooks/toasts.hooks';
 import { datePipe } from '../../../utils/pipes';
 import ErrorPage from '../../ErrorPage';
-import { NERButton } from '../../../components/NERButton';
-import { useState } from 'react';
 import AdminToolTable from '../AdminToolTable';
-import { useGetAllManufacturers } from '../../../hooks/bom.hooks';
 import CreateManufacturerModal from './CreateManufacturerFormModal';
+import ManufacturerDeleteModal from './ManufacturerDeleteModal';
+import { useState } from 'react';
+
+interface ManufacturerDeleteButtonProps {
+  name: string;
+  onDelete: (name: string) => void;
+}
 
 const ManufacturerTable: React.FC = () => {
   const {
-    data: manufactureres,
-    isLoading: manufactureresIsLoading,
+    data: manufacturers,
+    isLoading: manufacturersIsLoading,
     isError: manufacturersIsError,
     error: manufacturersError
   } = useGetAllManufacturers();
   const [createModalShow, setCreateModalShow] = useState<boolean>(false);
+  const { mutateAsync } = useDeleteManufacturer();
+  const toast = useToast();
 
-  if (!manufactureres || manufactureresIsLoading) {
+  if (!manufacturers || manufacturersIsLoading) {
     return <LoadingIndicator />;
   }
   if (manufacturersIsError) {
     return <ErrorPage message={manufacturersError?.message} />;
   }
 
-  const manufacturersTableRows = manufactureres.map((manufacturer) => (
+  const handleDeleteManufacturer = (manufacturerName: string) => {
+    try {
+      mutateAsync({ manufacturerName: manufacturerName });
+      toast.success(`Manufacturer: ${manufacturerName} Deleted Successfully!`);
+    } catch (error) {
+      toast.error(`Error Deleting Manufacturer!`);
+    }
+  };
```

Comment:
The code should be like this, I think what's happening is the mutateAsync is getting passed before it actually returns an error because we're not awaiting it, so the try/ catch doesn't even have a chance to catch it:
```
  const handleDeleteManufacturer = async (manufacturerName: string) => {
    try {
      await mutateAsync({ manufacturerName: manufacturerName });
      toast.success(`Manufacturer: ${manufacturerName} Deleted Successfully!`);
    } catch (error: unknown) {
      if (error instanceof Error) {
        toast.error(error.message);
      }
    }
  };
```